app/controllers/api/v1/products_controller.rb

```diff
@@ -37,6 +37,14 @@ def destroy
         head :no_content
       end
 
+      def assign_product_details
+        if @product.update(product_details: [product_params[:product_details].first])
```

Comment:
This method is replacing the already existing `product_details`, which is not the expected behavior (You should append to the existing `product_details`.
Also this is only taking the first element of the newly provided array. Therefore at the end, there will be always one `product_details` available.